src/library-authoring/EmptyStates.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import {
+  Button, Stack,
+} from '@openedx/paragon';
+import { Add } from '@openedx/paragon/icons';
+
+import messages from './messages';
+
+export const NoComponents = () => (
+  <Stack direction="horizontal" gap={3} className="mt-6 justify-content-center">
+    <FormattedMessage {...messages.noComponents} />
+    <Button iconBefore={Add}>
+      <FormattedMessage {...messages.addComponent} />
```

Comment:
This isn't specific to this code, but I've seen this pattern a lot and I don't personally see the value in it. 

To me the advantage of using FormattedMessage is that you can see the actual text in the react code instead of looking at the messages file. In this case it's better to use `intl.formatMessage` instead. `FormattedMessage` just uses `useIntl` internally anyway.

Reply:
Are you saying to write `{ intl.formatMessage(messages.addComponent) }` ? That's totally fine, but requires changing this function to have a statement body to be able to include `const intl = useIntl()` in scope, so writing it this way uses one less line of code, and I think is just as simple.